### PR TITLE
Feat/discover button api setup

### PR DESCRIPTION
**Pull request by sdmisra**
# PR Template

- [] I have ran all tests (if applicable) and they are passing. 

### Types of changes made?

- []  testing
- []  documentation
- [x]  javascript
- [x]  functionality

### What does this PR do?

- Adds functionality
Enables discover button to access the undiscovered worlds endpoint.
- Updates JS for
WelcomePage, apicalls

### Notes

catch of apicall will need to be replaced with our error handling routes

console.logs intentionally left in to help with final steps of debugging or refactoring

### CODE REVIEW

File: src/Components/WelcomePage/WelcomePage.js [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+  const history = useHistory()
+  const displayedWorlds = useSelector((state) => state.root.discoveredWorlds)
+
+  const discoverNewWorld = ()=> {
+    getRandomWorldData()
+    .then(data=> {
+      const addOne = [...displayedWorlds, data]
+      dispatch(getDiscoveredWorlds(addOne))
+      // console.log(data.id)
+      history.push(`/world/${data.id}`)
```
Comment (priscillaapaxton) [inline src/Components/WelcomePage/WelcomePage.js:19]: can we discuss further the use of the useHistory method? I am on board with the dispatch of getDiscoveredWorlds to eventually update the discoveredWorlds global state, but why not just make the button a navlink to the path of world/${data.id}? we can discuss more in person!

Comment (sdmisra) [reply src/Components/WelcomePage/WelcomePage.js:19]: We can definitely discuss it further - i only resorted to using the useHistory way when i failed to make a Redirect router component aim at that same sort of interpolated path

If we can make it work with a navlink, link, redirect, etc, i am down to switch it up!

Comment (Adam-Meza) [reply src/Components/WelcomePage/WelcomePage.js:19]: yeah i agree with priscilla here. i think a navlink would work better

**Review by Adam-Meza — APPROVED** [review]
Looks good! if using the history object is working im ok with merging this

**Pull request merged**